### DISCUSSION TITLE
About using the ask function of Ask-and-Tell Interface multiple times without the tell function

### DISCUSSION THREAD

**Discussion by yknoww** [optuna/optuna#4815] [Q&A]
When I run the code below, I got different suggested values for each.
**I expected to get the same value, why do I get different values?**

> -0.165955990594852
> 0.4406489868843162

```
import optuna
from optuna.samplers import TPESampler

study = optuna.create_study(sampler=TPESampler(seed=1))
trial = study.ask()
x1 = trial.suggest_float("x", -1, 1)
print(x1)
trial = study.ask()
x2 = trial.suggest_float("x", -1, 1)
print(x2)
```

**Comment by nzw0301** [comment]
In my understanding, samplers have `numpy.random.RandomState` to generate params, `x` in this example. By calling `suggest_float`, `RandomState`'s state changes, as a result, we have different values as you demonstrate.
